internal/lib/lib.go

```diff
@@ -980,3 +981,104 @@ func IsAviLBDefaultIngressClassWithClient(kc kubernetes.Interface) (string, bool
 	utils.AviLog.Debugf("IngressClass with controller ako.vmware.com/avi-lb not found in the cluster")
 	return "", false
 }
+
+func GetAviSecretWithRetry(kc kubernetes.Interface, retryCount int) (*v1.Secret, error) {
+	var aviSecret *v1.Secret
+	var err error
+	for retry := 0; retry < retryCount; retry++ {
+		aviSecret, err = kc.CoreV1().Secrets(AviNS).Get(context.TODO(), AviSecret, metav1.GetOptions{})
+		if err == nil {
+			return aviSecret, nil
+		}
+		utils.AviLog.Warnf("Failed to get avi-secret, retry count:%d, err: %+v", retry, err)
+	}
+	return nil, err
+}
+
+func UpdateAviSecretWithRetry(kc kubernetes.Interface, aviSecret *v1.Secret, retryCount int) error {
+	var err error
+	for retry := 0; retry < retryCount; retry++ {
+		_, err = kc.CoreV1().Secrets(AviNS).Update(context.TODO(), aviSecret, metav1.UpdateOptions{})
+		if err == nil {
+			return nil
+		}
+		utils.AviLog.Warnf("Failed to update avi-secret, retry count:%d, err: %+v", retry, err)
+	}
+	return err
+}
+
+func RefreshAuthToken(kc kubernetes.Interface) {
+	retryCount := 5
+	ctrlProp := utils.SharedCtrlProp().GetAllCtrlProp()
+	ctrlUsername := ctrlProp[utils.ENV_CTRL_USERNAME]
+	ctrlAuthToken := ctrlProp[utils.ENV_CTRL_AUTHTOKEN]
+	ctrlIpAddress := os.Getenv(utils.ENV_CTRL_IPADDRESS)
+
+	aviClient := utils.NewAviRestClientWithToken(ctrlIpAddress, ctrlUsername, ctrlAuthToken)
+	if aviClient == nil {
+		utils.AviLog.Errorf("Failed to initialize AVI client")
+		return
+	}
+	userTokensListResp, err := utils.GetAuthTokenWithRetry(aviClient, retryCount)
+	if err != nil {
+		utils.AviLog.Errorf("Failed to get existing tokens from controller, err: %+v", err)
+		return
+	}
+	oldTokenID, refresh, err := utils.GetTokenFromRestObj(userTokensListResp, ctrlAuthToken)
+	if err != nil {
+		utils.AviLog.Errorf("Failed to find token on controller, err: %+v", err)
+		return
+	}
+	if !refresh {
+		utils.AviLog.Infof("Skipping AuthToken Refresh")
```

Comment:
Shouldn't we return from here? If we are skipping auth token refresh, then we are still continuing to the next line.

Reply:
added the missing return, thanks